clang/include/clang/Basic/Builtins.td

```diff
@@ -4272,6 +4272,12 @@ def MoveIfNsoexcept : CxxLibBuiltin<"utility"> {
   let Namespace = "std";
 }
 
+def Invoke : Builtin {
+  let Spellings = ["__builtin_invoke"];
+  let Attributes = [CustomTypeChecking, Constexpr];
```

Comment:
There are no changes to `ExprConstant.cpp` - is this actually tested in a constexpr context?

Reply:
We should still add test coverage to `clang/test/SemaCXX` for it though; we don't want to have to wait for us to break libc++ tests to know we've regressed anything here.